// ngx-extended-pdf-viewer/esm5/lib/secondary-toolbar/pdf-secondary-toolbar/pdf-secondary-toolbar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter, TemplateRef, ElementRef, HostListener } from '@angular/core';
var PdfSecondaryToolbarComponent = /** @class */ (function () {
    function PdfSecondaryToolbarComponent(element) {
        this.element = element;
        this.showPresentationModeButton = true;
        this.showOpenFileButton = true;
        this.showPrintButton = true;
        this.showDownloadButton = true;
        this.showBookmarkButton = true;
        this.showPagingButtons = true;
        this.showRotateButton = true;
        this.showHandToolButton = true;
        this.showScrollingButton = true;
        this.showSpreadButton = true;
        this.showPropertiesButton = true;
        this.spreadChange = new EventEmitter();
        this.secondaryMenuIsEmpty = new EventEmitter();
    }
    /**
     * @param {?} newSpread
     * @return {?}
     */
    PdfSecondaryToolbarComponent.prototype.onSpreadChange = /**
     * @param {?} newSpread
     * @return {?}
     */
    function (newSpread) {
        this.spreadChange.emit(newSpread);
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    PdfSecondaryToolbarComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        var _this = this;
        setTimeout((/**
         * @return {?}
         */
        function () { return _this.checkVisibility(); }));
    };
    /**
     * @return {?}
     */
    PdfSecondaryToolbarComponent.prototype.onResize = /**
     * @return {?}
     */
    function () {
        var _this = this;
        setTimeout((/**
         * @return {?}
         */
        function () { return _this.checkVisibility(); }));
    };
    /**
     * @return {?}
     */
    PdfSecondaryToolbarComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        setTimeout((/**
         * @return {?}
         */
        function () { return _this.checkVisibility(); }));
    };
    /**
     * @return {?}
     */
    PdfSecondaryToolbarComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        setTimeout((/**
         * @return {?}
         */
        function () { return _this.checkVisibility(); }));
    };
    /**
     * @return {?}
     */
    PdfSecondaryToolbarComponent.prototype.checkVisibility = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var visibleButtons = 0;
        /** @type {?} */
        var e = (/** @type {?} */ (this.element.nativeElement));
        /** @type {?} */
        var f = e.children.item(0);
        if (f) {
            /** @type {?} */
            var g = f.children.item(0);
            if (g && g instanceof HTMLElement) {
                visibleButtons = this.checkVisibilityRecursively(g);
            }
        }
        this.secondaryMenuIsEmpty.emit(visibleButtons === 0);
    };
    /**
     * @private
     * @param {?} e
     * @return {?}
     */
    PdfSecondaryToolbarComponent.prototype.checkVisibilityRecursively = /**
     * @private
     * @param {?} e
     * @return {?}
     */
    function (e) {
        if (e.style.display === 'none') {
            return 0;
        }
        if (e.classList.contains('hidden')) {
            return 0;
        }
        if (e.classList.contains('invisible')) {
            return 0;
        }
        /** @type {?} */
        var style = window.getComputedStyle(e);
        if (style.display === 'none') {
            return 0;
        }
        if (e instanceof HTMLButtonElement || e instanceof HTMLAnchorElement) {
            return 1;
        }
        /** @type {?} */
        var count = 0;
        /** @type {?} */
        var children = e.children;
        if (children && children.length) {
            for (var i = 0; i < children.length && count === 0; i++) {
                /** @type {?} */
                var child = children.item(i);
                if (child && child instanceof HTMLElement) {
                    count += this.checkVisibilityRecursively(child);
                }
            }
        }
        return count;
    };
    PdfSecondaryToolbarComponent.decorators = [
        { type: Component, args: [{
                    selector: 'pdf-secondary-toolbar',
                    template: "<ng-container [ngTemplateOutlet]=\"customSecondaryToolbar ? customSecondaryToolbar : defaultSecondaryToolbar\"> </ng-container>\n\n<ng-template #defaultSecondaryToolbar>\n  <div\n    id=\"secondaryToolbar\"\n    class=\"secondaryToolbar hidden doorHangerRight\"\n    [style.top]=\"secondaryToolbarTop\"\n    [style.transform]=\"'scale(' + mobileFriendlyZoomScale + ')'\"\n    [style.transformOrigin]=\"'right top'\"\n  >\n    <div id=\"secondaryToolbarButtonContainer\">\n      <button\n        type=\"button\"\n        type=\"button\"\n        id=\"secondaryPresentationMode\"\n        [class.invisible]=\"!showPresentationModeButton\"\n        class=\"secondaryToolbarButton visibleLargeView\"\n        title=\"Switch to Presentation Mode\"\n        tabindex=\"51\"\n        data-l10n-id=\"presentation_mode\"\n      >\n        <svg style=\"width:22px;height:22px\" viewBox=\"0 0 24 24\">\n          <path fill=\"currentColor\" d=\"M5,5H10V7H7V10H5V5M14,5H19V10H17V7H14V5M17,14H19V19H14V17H17V14M10,17V19H5V14H7V17H10Z\" />\n        </svg>\n        <span data-l10n-id=\"presentation_mode_label\">Presentation Mode</span>\n      </button>\n\n      <button\n        type=\"button\"\n        type=\"button\"\n        id=\"secondaryOpenFile\"\n        [class.invisible]=\"!showOpenFileButton\"\n        class=\"secondaryToolbarButton visibleLargeView\"\n        title=\"Open File\"\n        tabindex=\"52\"\n        data-l10n-id=\"open_file\"\n      >\n        <svg style=\"width:22px;height:22px\" viewBox=\"0 0 24 24\">\n          <path fill=\"currentColor\" d=\"M14,2L20,8V20A2,2 0 0,1 18,22H6A2,2 0 0,1 4,20V4A2,2 0 0,1 6,2H14M18,20V9H13V4H6V20H18M12,12L16,16H13.5V19H10.5V16H8L12,12Z\" />\n        </svg>\n        <span data-l10n-id=\"open_file_label\">Open</span>\n      </button>\n\n      <button\n        type=\"button\"\n        type=\"button\"\n        id=\"secondaryPrint\"\n        class=\"secondaryToolbarButton visibleMediumView\"\n        [class.invisible]=\"!showPrintButton\"\n        title=\"Print\"\n        tabindex=\"53\"\n        data-l10n-id=\"print\"\n      >\n      <svg style=\"width:22px;height:22px\" viewBox=\"0 0 24 24\">\n        <path fill=\"currentColor\" d=\"M18,3H6V7H18M19,12A1,1 0 0,1 18,11A1,1 0 0,1 19,10A1,1 0 0,1 20,11A1,1 0 0,1 19,12M16,19H8V14H16M19,8H5A3,3 0 0,0 2,11V17H6V21H18V17H22V11A3,3 0 0,0 19,8Z\" />\n    </svg>\n        <span data-l10n-id=\"print_label\">Print</span>\n      </button>\n\n      <button\n        type=\"button\"\n        type=\"button\"\n        id=\"secondaryDownload\"\n        class=\"secondaryToolbarButton visibleMediumView\"\n        [class.invisible]=\"!showDownloadButton\"\n        title=\"Download\"\n        tabindex=\"54\"\n        data-l10n-id=\"download\"\n      >\n        <svg style=\"width:22px;height:22px\" viewBox=\"0 0 24 24\">\n          <path fill=\"currentColor\" d=\"M14,2L20,8V20A2,2 0 0,1 18,22H6A2,2 0 0,1 4,20V4A2,2 0 0,1 6,2H14M18,20V9H13V4H6V20H18M12,19L8,15H10.5V12H13.5V15H16L12,19Z\" />\n        </svg>\n        <span data-l10n-id=\"download_label\">Download</span>\n      </button>\n\n      <a\n        href=\"#\"\n        id=\"secondaryViewBookmark\"\n        class=\"secondaryToolbarButton bookmark visibleSmallView\"\n        [class.invisible]=\"!showBookmarkButton\"\n        title=\"Current view (copy or open in new window)\"\n        tabindex=\"55\"\n        data-l10n-id=\"bookmark\"\n      >\n        <svg style=\"width:22px;height:22px\" viewBox=\"0 0 24 24\">\n          <path fill=\"currentColor\" d=\"M17,18L12,15.82L7,18V5H17M17,3H7A2,2 0 0,0 5,5V21L12,18L19,21V5C19,3.89 18.1,3 17,3Z\" />\n        </svg>\n        <span data-l10n-id=\"bookmark_label\">Current View</span>\n      </a>\n\n      <div class=\"horizontalToolbarSeparator visibleLargeView\"></div>\n\n      <button\n        type=\"button\"\n        type=\"button\"\n        [class.invisible]=\"!showPagingButtons\"\n        id=\"firstPage\"\n        class=\"secondaryToolbarButton firstPage\"\n        title=\"Go to First Page\"\n        tabindex=\"56\"\n        data-l10n-id=\"first_page\"\n      >\n        <span data-l10n-id=\"first_page_label\">Go to First Page</span>\n      </button>\n      <button\n        type=\"button\"\n        type=\"button\"\n        [class.invisible]=\"!showPagingButtons\"\n        id=\"lastPage\"\n        class=\"secondaryToolbarButton lastPage\"\n        title=\"Go to Last Page\"\n        tabindex=\"57\"\n        data-l10n-id=\"last_page\"\n      >\n        <span data-l10n-id=\"last_page_label\">Go to Last Page</span>\n      </button>\n\n      <div [class.invisible]=\"!showRotateButton\" class=\"horizontalToolbarSeparator\"></div>\n\n      <button\n        type=\"button\"\n        type=\"button\"\n        [class.invisible]=\"!showRotateButton\"\n        id=\"pageRotateCw\"\n        class=\"secondaryToolbarButton rotateCw\"\n        title=\"Rotate Clockwise\"\n        tabindex=\"58\"\n        data-l10n-id=\"page_rotate_cw\"\n      >\n        <span data-l10n-id=\"page_rotate_cw_label\">Rotate Clockwise</span>\n      </button>\n      <button\n        type=\"button\"\n        type=\"button\"\n        [class.invisible]=\"!showRotateButton\"\n        id=\"pageRotateCcw\"\n        class=\"secondaryToolbarButton rotateCcw\"\n        title=\"Rotate Counterclockwise\"\n        tabindex=\"59\"\n        data-l10n-id=\"page_rotate_ccw\"\n      >\n        <span data-l10n-id=\"page_rotate_ccw_label\">Rotate Counterclockwise</span>\n      </button>\n\n      <div class=\"horizontalToolbarSeparator\" [class.invisible]=\"!showHandToolButton\"></div>\n\n      <button\n        type=\"button\"\n        type=\"button\"\n        [class.invisible]=\"!showHandToolButton\"\n        id=\"cursorSelectTool\"\n        class=\"secondaryToolbarButton selectTool toggled\"\n        title=\"Enable Text Selection Tool\"\n        tabindex=\"60\"\n        data-l10n-id=\"cursor_text_select_tool\"\n      >\n        <span data-l10n-id=\"cursor_text_select_tool_label\">Text Selection Tool</span>\n      </button>\n      <button\n        type=\"button\"\n        type=\"button\"\n        [class.invisible]=\"!showHandToolButton\"\n        id=\"cursorHandTool\"\n        class=\"secondaryToolbarButton handTool\"\n        title=\"Enable Hand Tool\"\n        tabindex=\"61\"\n        data-l10n-id=\"cursor_hand_tool\"\n      >\n        <span data-l10n-id=\"cursor_hand_tool_label\">Hand Tool</span>\n      </button>\n\n      <div [class.invisible]=\"!showScrollingButton\" class=\"horizontalToolbarSeparator\"></div>\n\n      <button\n        type=\"button\"\n        type=\"button\"\n        [class.invisible]=\"!showScrollingButton\"\n        id=\"scrollVertical\"\n        class=\"secondaryToolbarButton scrollVertical toggled\"\n        title=\"Use Vertical Scrolling\"\n        tabindex=\"62\"\n        data-l10n-id=\"scroll_vertical\"\n      >\n        <span data-l10n-id=\"scroll_vertical_label\">Vertical Scrolling</span>\n      </button>\n      <button\n        type=\"button\"\n        [class.invisible]=\"!showScrollingButton\"\n        id=\"scrollHorizontal\"\n        class=\"secondaryToolbarButton scrollHorizontal\"\n        title=\"Use Horizontal Scrolling\"\n        tabindex=\"63\"\n        data-l10n-id=\"scroll_horizontal\"\n      >\n        <span data-l10n-id=\"scroll_horizontal_label\">Horizontal Scrolling</span>\n      </button>\n      <button\n        type=\"button\"\n        [class.invisible]=\"!showScrollingButton\"\n        id=\"scrollWrapped\"\n        class=\"secondaryToolbarButton scrollWrapped\"\n        title=\"Use Wrapped Scrolling\"\n        tabindex=\"64\"\n        data-l10n-id=\"scroll_wrapped\"\n      >\n        <span data-l10n-id=\"scroll_wrapped_label\">Wrapped Scrolling</span>\n      </button>\n\n      <div [class.invisible]=\"!showSpreadButton\" class=\"horizontalToolbarSeparator\"></div>\n\n      <button\n        type=\"button\"\n        [class.invisible]=\"!showSpreadButton\"\n        id=\"spreadNone\"\n        class=\"secondaryToolbarButton spreadNone toggled\"\n        title=\"Do not join page spreads\"\n        tabindex=\"65\"\n        data-l10n-id=\"spread_none\"\n        (click)=\"onSpreadChange('off')\"\n      >\n        <span data-l10n-id=\"spread_none_label\">No Spreads</span>\n      </button>\n      <button\n        type=\"button\"\n        [class.invisible]=\"!showSpreadButton\"\n        id=\"spreadOdd\"\n        class=\"secondaryToolbarButton spreadOdd\"\n        title=\"Join page spreads starting with odd-numbered pages\"\n        tabindex=\"66\"\n        data-l10n-id=\"spread_odd\"\n        (click)=\"onSpreadChange('odd')\"\n      >\n        <span data-l10n-id=\"spread_odd_label\">Odd Spreads</span>\n      </button>\n      <button\n        type=\"button\"\n        [class.invisible]=\"!showSpreadButton\"\n        id=\"spreadEven\"\n        class=\"secondaryToolbarButton spreadEven\"\n        title=\"Join page spreads starting with even-numbered pages\"\n        tabindex=\"67\"\n        data-l10n-id=\"spread_even\"\n        (click)=\"onSpreadChange('even')\"\n      >\n        <span data-l10n-id=\"spread_even_label\">Even Spreads</span>\n      </button>\n\n      <div [class.invisible]=\"!showPropertiesButton\" class=\"horizontalToolbarSeparator spreadModeButtons\"></div>\n\n      <button\n        type=\"button\"\n        [class.invisible]=\"!showPropertiesButton\"\n        id=\"documentProperties\"\n        class=\"secondaryToolbarButton documentProperties\"\n        title=\"Document Properties\u2026\"\n        tabindex=\"68\"\n        data-l10n-id=\"document_properties\"\n      >\n        <span data-l10n-id=\"document_properties_label\">Document Properties\u2026</span>\n      </button>\n    </div>\n  </div>\n</ng-template>\n",
                    styles: ["svg{position:absolute;display:inline-block;top:0;left:0}"]
                }] }
    ];
    /** @nocollapse */
    PdfSecondaryToolbarComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    PdfSecondaryToolbarComponent.propDecorators = {
        customSecondaryToolbar: [{ type: Input }],
        secondaryToolbarTop: [{ type: Input }],
        mobileFriendlyZoomScale: [{ type: Input }],
        showPresentationModeButton: [{ type: Input }],
        showOpenFileButton: [{ type: Input }],
        showPrintButton: [{ type: Input }],
        showDownloadButton: [{ type: Input }],
        showBookmarkButton: [{ type: Input }],
        showPagingButtons: [{ type: Input }],
        showRotateButton: [{ type: Input }],
        showHandToolButton: [{ type: Input }],
        showScrollingButton: [{ type: Input }],
        showSpreadButton: [{ type: Input }],
        showPropertiesButton: [{ type: Input }],
        spreadChange: [{ type: Output }],
        secondaryMenuIsEmpty: [{ type: Output }],
        onResize: [{ type: HostListener, args: ['window:resize',] }]
    };
    return PdfSecondaryToolbarComponent;
}());
export { PdfSecondaryToolbarComponent };
if (false) {
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.customSecondaryToolbar;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.secondaryToolbarTop;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.mobileFriendlyZoomScale;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.showPresentationModeButton;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.showOpenFileButton;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.showPrintButton;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.showDownloadButton;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.showBookmarkButton;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.showPagingButtons;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.showRotateButton;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.showHandToolButton;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.showScrollingButton;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.showSpreadButton;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.showPropertiesButton;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.spreadChange;
    /** @type {?} */
    PdfSecondaryToolbarComponent.prototype.secondaryMenuIsEmpty;
    /**
     * @type {?}
     * @private
     */
    PdfSecondaryToolbarComponent.prototype.element;
}
